src/OrbitGl/FrameTrack.cpp

```diff
@@ -52,7 +52,7 @@ float FrameTrack::GetMaximumBoxHeight() const {
   if (scale_factor == 0.f) {
     return 0.f;
   }
-  return scale_factor * box_height_ / box_height_normalizer;
+  return scale_factor * GetBoxHeight() / box_height_normalizer;
```

Comment:
The is no need to keep both `scale_factor` and `box_height_normalizer`. You can change to do the following
```cpp
float scale_factor = is_collapsed ? 1.f : GetMaximumScaleFactor();
return scale_factor * GetBoxHeight();
```

By doing this we can also remove the additional check of `scale_factor == 0.f`.

Reply:
Thanks for catching and pointing this! I'm already refactoring here because it was very confused to me to read it. I will change the code as you said but also GetMaximumScaleFactor shouldn't be used here. A PR is coming soon.